lib/modules/manager/nix/artifacts.ts

```diff
@@ -0,0 +1,81 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { readLocalFile } from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  config,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  const lockFileName = packageFileName.replace(regEx(/\.nix$/), '.lock');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    logger.debug('No flake.lock found');
+    return null;
+  }
+
+  let cmd: string;
+
+  if (config.isLockFileMaintenance) {
+    cmd = `nix \
+    --extra-experimental-features nix-command \
+    --extra-experimental-features flakes \
+    flake update`;
+  } else {
+    const inputs = updatedDeps
+      .map(({ depName }) => depName)
+      .filter(is.nonEmptyStringAndNotWhitespace)
+      .map((depName) => `--update-input ${depName}`)
```

Comment:
```suggestion
      .map((depName) => `--update-input ${quote(depName)}`)
```

needs to be quoted (security)